File: LeetCode/Java/11-two-pointers-盛最多水的容器.java
/*
 * @lc app=leetcode.cn id=11 lang=java
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
class Solution {
    public int maxArea(int[] height) {
        int i = 0, j = height.length - 1;
        int maxArea = 0;

        while (i < j) {
            int minHeight = height[i] < height[j] ? height[i++] : height[j--];
            int area = (j - i + 1) * minHeight;
            maxArea = Math.max(maxArea, area);
        }

        return maxArea;
    }
}
// @lc code=end

File: LeetCode/C/11-two-pointers-盛最多水的容器.cpp
/*
 * @lc app=leetcode.cn id=11 lang=cpp
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
class Solution {
  public:
    int maxArea(vector<int> &height) {
        int res = 0;
        int l = 0, r = height.size() - 1;
        while (l < r) {
            int h = height[l] < height[r] ? height[l++] : height[r--];
            res = max(res, h * (r - l + 1));
        }
        return res;
    }
};
// @lc code=end
